### PR TITLE
Add contributing instructions.

### PR DESCRIPTION
**Pull request by daspecster**
Hello!
I noticed from #521, that you guys were looking for something like this to cut the next release.

Let me know if there is anything you want to change or add!

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -1,4 +1,72 @@
...
+Step 1: Forking pycodestyle for editing
+---------------------------------------
+
+Fork the pycodestyle repository on github. This will add
+pycodestyle to your github account. You will push your changes to your
+fork and then make pull requests into the official pycodestyle repository.
+
+Github has an excellent `guide`_ that has screenshots on how to do this.
+
+Next, clone your fork of the pycodestyle repository to your system for
```
Comment (IanLee1521) [inline CONTRIBUTING.rst]: Was this meant to be Step 2 ? If not, you just need to renumber steps 3 and 4 below to 2 and 3.

**Comment by IanLee1521** [comment]
This looks great, thank you! Just one comment on step numbering, then should be good to merge.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -1,4 +1,72 @@
...
+Contributing to pycodestyle
 ===============================
 
-Please see the `developer notes <https://pycodestyle.readthedocs.io/en/latest/developer.html>`_
+When contributing to pycodestyle, please observe our `Code of Conduct`_.
+
+Step 1: Forking pycodestyle for editing
+---------------------------------------
+
+Fork the pycodestyle repository on github. This will add
```
Comment (IanLee1521) [inline CONTRIBUTING.rst]: I usually prefer to spell it `GitHub` rather than `github` but that's just me.

Comment (sigmavirus24) [inline CONTRIBUTING.rst]: Well, given "GitHub" is it's actual name (as trademarked and branded) that makes sense to me. It's also how I prefer to spell it.

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -1,4 +1,72 @@
-Contributing to ``pycodestyle``
+Contributing to pycodestyle
 ===============================
```
Comment (sigmavirus24) [inline CONTRIBUTING.rst]: The length of the `=`s below should match the length of the title.

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -1,4 +1,72 @@
...
+When contributing to pycodestyle, please observe our `Code of Conduct`_.
+
+Step 1: Forking pycodestyle for editing
+---------------------------------------
+
+Fork the pycodestyle repository on github. This will add
+pycodestyle to your github account. You will push your changes to your
+fork and then make pull requests into the official pycodestyle repository.
+
+Github has an excellent `guide`_ that has screenshots on how to do this.
```
Comment (sigmavirus24) [inline CONTRIBUTING.rst]: Git**H**ub

**Comment by daspecster** [comment]
Thanks for catching those! Updates pushed.

**Comment by sigmavirus24** [comment]
@daspecster you didn't quite get the capitalization right yet. Capital G **and** H in GitHub.

```
GitHub # good
Github  # bad
``

**Comment by daspecster** [comment]
Got it! Thanks!

Squashed the commit as well.

**Comment by sigmavirus24** [comment]
Thanks @daspecster! ✨ 🍛 ✨

**Pull request merged**